tests/diff/float_values_test.go

```diff
@@ -76,13 +164,65 @@ func TestFloatValues(t *testing.T) {
 		}
 	})
 
-	// TODO https://github.com/FerretDB/dance/issues/266
-	/*t.Run("Update", func(t *testing.T) {
+	t.Run("FindAndModify", func(t *testing.T) {
+		t.Parallel()
 
-	})*/
+		for name, tc := range map[string]struct {
+			filter   bson.D
+			update   bson.D
+			expected mongo.CommandError
+		}{
+			"NaN": {
+				filter: bson.D{{"_id", "1"}},
+				update: bson.D{{"$set", bson.D{{"foo", math.NaN()}}}},
+				expected: mongo.CommandError{
+					Code: 2,
+					Name: "BadValue",
+					Message: `wire.OpMsg.Document: validation failed for { findA` +
+						`ndModify: "findAndModify-NaN", query: { _id: "1" }, upd` +
+						`ate: { $set: { foo: nan.0 } }, $db: "testfloatvalues" } with: NaN is not supported`,
```

Comment:
Let's not split words inside the message to make it more readable
```suggestion
					Message: `wire.OpMsg.Document: validation failed for ` +
						`{ findAndModify: "findAndModify-NaN", query: { _id: "1" }, ` +
						`update: { $set: { foo: nan.0 } }, $db: "testfloatvalues" } with: NaN is not supported`,
```

Reply:
I agree! I will fix this.